src/UI/Component/ViewControl/Factory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/* Copyright (c) 2016 Timon Amstutz <timon.amstutz@ilub.unibe.ch> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\ViewControl;
+/**
+ * This is how the factory for UI elements looks.
+ */
+interface Factory {
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *      Mode View Controls enable the switching between different aspects of some data. The different modes are mutually
+	 *      exclusive and can therefore not be activated at once.
+	 *   composition: >
+	 *      Mode View Controls are composed of Buttons switching between active and inactive states.
+	 *   effect: >
+	 *      Clicking on an inactive Button turns this button active and all other inactive. Clicking on an active button
+	 *      has no effect.
+	 *
+	 * rules:
+	 *   usage:
+	 *      1: Exactly one Button MUST always be active.
+	 *   accessibility:
+	 *      1: The HTML container enclosing the buttons of the Mode View Control MUST cary the role-attribute "group".
+	 *      2: The HTML container enclosing the buttons of the Mode View Control MUST set an aria-label describing the element. Eg. "Mode View Control"
+	 *      3: The Buttons of the Mode View Control MUST set an aria-label clearly describing what the button shows if clicked. E.g. "List View", "Month View", ...
```

Comment:
I don't know how far we want to go with this assistive attributes (aria-labels and so on). Can we add this attributes directly in the Button? 
@nhaagen had the same issue with the UI Tag #557